react/__tests__/pickInstallments.test.ts

```diff
@@ -40,3 +43,78 @@ describe('pickInstallments', () => {
     expect(pickedInstallments[index].NumberOfInstallments).toBe(15)
   })
 })
+
+describe('pickMaxInstallmentsOption', () => {
+  it('should pick the installments plan with the highest NumberOfInstallments', () => {
+    const pickedInstallmentsPlan = pickMaxInstallmentsOption(
+      installmentsListMastercardMax
+    )
+
+    expect(pickedInstallmentsPlan.NumberOfInstallments).toBe(15)
+    expect(pickedInstallmentsPlan.PaymentSystemName).toBe('Mastercard')
+  })
+
+  it('should pick the installments plan with the highest NumberOfInstallments, from a certain payment system', () => {
+    const pickedInstallmentsPlan = pickMaxInstallmentsOption(installmentsList, {
+      paymentSystemName: 'Customer Credit',
+    })
+
+    expect(pickedInstallmentsPlan.NumberOfInstallments).toBe(3)
+    expect(pickedInstallmentsPlan.PaymentSystemName).toBe('Customer Credit')
+  })
+
+  it('should pick an installments plan with a certain number of installments', () => {
+    const pickedInstallmentsPlan = pickMaxInstallmentsOption(
+      installmentsListMastercardMax,
+      {
+        installmentsQuantity: 14,
+      }
+    )
+
+    expect(pickedInstallmentsPlan.NumberOfInstallments).toBe(14)
+  })
+
+  it('should pick the installments plan with the highest NumberOfInstallments, using all filtering options', () => {
+    const pickedInstallmentsPlan = pickMaxInstallmentsOption(installmentsList, {
+      paymentSystemName: 'Customer Credit',
+      installmentsQuantity: 2,
+    })
+
+    expect(pickedInstallmentsPlan.NumberOfInstallments).toBe(2)
+    expect(pickedInstallmentsPlan.PaymentSystemName).toBe('Customer Credit')
+  })
+})
+
+describe('pickMaxInstallmentsOptionWithNoInterest', () => {
+  it('should pick the installments plan with the highest NumberOfInstallments and no interest', () => {
+    const pickedInstallmentsPlan = pickMaxInstallmentsOptionWithNoInterest(
+      installmentsListMastercardMax
+    )
+
+    expect(pickedInstallmentsPlan.NumberOfInstallments).toBe(14)
+    expect(pickedInstallmentsPlan.PaymentSystemName).toBe('Visa')
+    expect(pickedInstallmentsPlan.InterestRate).toBe(0)
+  })
+
+  it('should pick the installments plan with the second highest NumberOfInstallments and no interest, due to filtering rules', () => {
+    const pickedInstallmentsPlan = pickMaxInstallmentsOptionWithNoInterest(
+      installmentsList,
+      {
+        paymentSystemName: 'Customer Credit',
+      }
+    )
+
+    expect(pickedInstallmentsPlan.NumberOfInstallments).toBe(2)
+    expect(pickedInstallmentsPlan.PaymentSystemName).toBe('Customer Credit')
+    expect(pickedInstallmentsPlan.InterestRate).toBe(0)
+  })
+
+  it("should pick the installments plan with the highest NumberOfInstallments overall, if there isn't an option with no interest", () => {
```

Comment:
I think this is an important information to put on README, so we make it clear to the reader that if there isn't a satisfying option we will return unfiltered.

Reply:
Agreed! For sure!